corehq/apps/enterprise/tests/test_enterprise_mobile_worker_settings.py

```diff
@@ -0,0 +1,216 @@
+import datetime
+
+from django.test import TestCase
+
+from corehq.apps.domain.shortcuts import create_domain
+from corehq.apps.enterprise.models import EnterpriseMobileWorkerSettings
+from corehq.apps.enterprise.tests.utils import (
+    get_enterprise_account,
+    add_domains_to_enterprise_account,
+    get_enterprise_software_plan,
+)
+from corehq.apps.es.tests.utils import es_test
+from corehq.apps.users.models import CommCareUser
+from corehq.elastic import get_es_new, send_to_elasticsearch
+from corehq.form_processor.utils import TestFormMetadata
+from corehq.pillows.mappings import XFORM_INDEX_INFO
+from corehq.pillows.mappings.user_mapping import USER_INDEX_INFO
+from corehq.pillows.user import transform_user_for_elasticsearch
+from corehq.util.elastic import ensure_index_deleted
+from corehq.util.test_utils import make_es_ready_form
+from dimagi.utils.dates import add_months_to_date
+from pillowtop.es_utils import initialize_index_and_mapping
+
+
+@es_test
+class TestEnterpriseMobileWorkerSettings(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.es = get_es_new()
+        initialize_index_and_mapping(cls.es, USER_INDEX_INFO)
+        initialize_index_and_mapping(cls.es, XFORM_INDEX_INFO)
+
+        today = datetime.datetime.utcnow()
+
+        one_year_ago = add_months_to_date(today.date(), -12)
+        enterprise_plan = get_enterprise_software_plan()
```

Comment:
Does this need to be cleaned up? I've had problems with tests that create software plans in the past where running them multiple times in a local dev environment would result in setup failures, possibly due to caching or some other persistent state outside of SQL.

Aside: [`TestCase.addClassCleanup()`](https://docs.python.org/3.9/library/unittest.html#unittest.TestCase.addClassCleanup) is more reliable (in case of partial setup failure) and easier to maintain for large `setUpClass` methods like this one.

Reply:
added